src/device-config.ts

```diff
@@ -0,0 +1,512 @@
+				}
+				const logValue = { SUPERVISOR_VPN_CONTROL: step.target };
+				if (!initial) {
+					this.logger.logConfigChange(logValue);
+				}
+				try {
+					await this.setVPNEnabled(step.target);
+					if (!initial) {
+						this.logger.logConfigChange(logValue, { success: true });
+					}
+				} catch (err) {
+					this.logger.logConfigChange(logValue, { err });
+					throw err;
+				}
+			},
+			setBootConfig: async step => {
+				const configBackend = await this.getConfigBackend();
+				if (!_.isObject(step.target)) {
+					throw new Error(
+						'Non-dictionary passed to DeviceConfig.setBootConfig',
+					);
+				}
+				await this.setBootConfig(configBackend, step.target as Dictionary<
+					string
+				>);
+			},
+		};
+	}
+
+	private async getConfigBackend() {
+		if (this.configBackend != null) {
+			return this.configBackend;
+		}
+		const dt = await this.config.get('deviceType');
+		if (!_.isString(dt)) {
+			throw new Error('Could not detect device type');
+		}
+
+		return (this.configBackend = configUtils.getConfigBackend(dt) || null);
+	}
+
+	public async setTarget(
+		target: Dictionary<string>,
+		trx?: Transaction,
+	): Promise<void> {
+		const db = trx != null ? trx : this.db.models.bind(this.db);
+
+		const formatted = await this.formatConfigKeys(target);
+		const confToUpdate = {
+			targetValues: JSON.stringify(formatted),
+		};
+		await db('deviceConfig').update(confToUpdate);
+	}
+
+	public async getTarget({ initial = false }: { initial?: boolean } = {}) {
+		const [unmanaged, [devConfig]] = await Promise.all([
+			this.config.get('unmanaged'),
+			this.db.models('deviceConfig').select('targetValues'),
+		]);
+
+		let conf: Dictionary<string>;
+		try {
+			conf = JSON.parse(devConfig.targetValues);
+		} catch (e) {
+			throw new Error(`Corrupted supervisor database! Error: ${e.message}`);
+		}
+		if (initial || conf.SUPERVISOR_VPN_CONTROL == null) {
+			conf.SUPERVISOR_VPN_CONTROL = 'true';
+		}
+		if (unmanaged && conf.SUPERVISOR_LOCAL_MODE == null) {
+			conf.SUPERVISOR_LOCAL_MODE = 'true';
+		}
+
+		_.each(DeviceConfig.configKeys, ({ envVarName, defaultValue }) => {
+			if (conf[envVarName] == null) {
+				conf[envVarName] = defaultValue;
+			}
+		});
+
+		return conf;
+	}
+
+	public async getCurrent() {
+		const conf = await this.config.getMany(
+			['deviceType'].concat(_.keys(DeviceConfig.configKeys)),
+		);
+
+		const configBackend = await this.getConfigBackend();
+
+		const [vpnStatus, bootConfig] = await Promise.all([
+			this.getVPNEnabled(),
+			this.getBootConfig(configBackend),
+		]);
+
+		const currentConf: Dictionary<string> = {
+			// TODO: Fix this mess of half strings half boolean values everywhere
+			SUPERVISOR_VPN_CONTROL: vpnStatus != null ? vpnStatus.toString() : 'true',
+		};
+
+		for (const key in DeviceConfig.configKeys) {
+			const { envVarName } = DeviceConfig.configKeys[key];
+			const value = conf[key] != null ? (conf[key] as string) : '';
```

Comment:
This would be better as:
```suggestion
			currentConf[envVarName] = conf[key] != null ? conf[key].toString() : '';
```
to avoid the need for the cast (which was incorrect anyway if we actually needed the `toString`)